compiler/rustc_middle/src/mir/interpret/pointer.rs

```diff
@@ -18,6 +18,11 @@ pub trait PointerArithmetic: HasDataLayout {
         self.data_layout().pointer_size
     }
 
+    #[inline(always)]
+    fn max_size_of_val(&self) -> Size {
+        Size::from_bytes(self.machine_isize_max())
```

Comment:
This is somewhat dubious since it violates a comment placed on `Size`. See https://github.com/rust-lang/rust/pull/80042#pullrequestreview-922498771.

Reply:
Yea... I think we should look into fixing this by getting rid of the bits function or at least making it panic.

Reply:
Well it was changed to no longer panic in https://github.com/rust-lang/rust/pull/80042, presumably for some binary size wins?

Reply:
All right, prepared a separate PR for that: https://github.com/rust-lang/rust/pull/95456

Reply:
https://github.com/rust-lang/rust/pull/95456 landed and I rebased, so this concern should be resolved.